storage/sql/crud.go

```diff
@@ -574,12 +606,22 @@ func (c *conn) UpdatePassword(email string, updater func(p storage.Password) (st
 	return c.ExecTx(func(tx *trans) error {
 		p, err := getPassword(tx, email)
 		if err != nil {
-			return err
+			// Let's see if it is a storage.Error, if so just pass it on
+			if _, ok := err.(storage.Error); ok {
+				return err
+			}
+			// Nope, let's make a generic storage.Error of it
+			return storage.Error{Code: storage.ErrStorageProviderInternalError, Details: err.Error()}
```

Comment:
Can we make this a helper method?
```go
func storage.EnsureStorageError(err error) error {
	if _, ok := err.(storage.Error); ok {
		return err
	}
	return storage.Error{Code: storage.ErrStorageProviderInternalError, Details: err.Error()}
}
```
and use that instead of this block? 😃 

If we throw in a 
```go
	if err == nil {
		return nil
	}
```
we can even simplify call sites some more with this, maybe?

Reply:
Yeah, there was definitely a lot of copy-pasta.  At first I was ambitious of having relevant error messages - which I think is still a good idea - but on the other hand there are so many places errors can happen that it just was way too much work for a single PR.

Having a helper function and using it a lot right now probably makes the most sense - we can always add improved error messages when we have time or a pressing reason.